pkgs/by-name/lr/lrcget/package.nix

```diff
@@ -0,0 +1,100 @@
+{ dbus
+, openssl
+, gtk3
+, webkitgtk
+, pkg-config
+, wrapGAppsHook3
+, fetchFromGitHub
+, buildNpmPackage
+, rustPlatform
+, lib
+, stdenv
+, copyDesktopItems
+, makeDesktopItem
+, alsa-lib
+}:
+let
+  pname = "lrcget";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "tranxuanthang";
+    repo = "lrcget";
+    rev = "${version}";
+    hash = "sha256-9Wtb8Y+cj1ec8vNxvvs3wso62TavNveHRh2Gfx7Br4w=";
+  };
+in
+rustPlatform.buildRustPackage rec {
+  inherit version src pname;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoHash = "sha256-nhED/fSexFfZ2MezBR3YW5wKsYklRg9HiA3lvClkYjI=";
+
+  frontend = buildNpmPackage {
+    inherit version src;
+    pname = "lrcget-ui";
+    # FIXME: This is a workaround, because we have an git dependency node_modules/lrc-kit contains install scripts
+    # but has no lockfile, which is something that will probably break.
+    forceGitDeps = true;
+    distPhase = "true";
+    dontInstall = true;
+    # To fix `npm ERR! Your cache folder contains root-owned files`
+    makeCacheWritable = true;
+
+    npmDepsHash = "sha256-qQ5UMO3UuD6IvUveTRF35qTlGq5PMbxp1Q4UroDqVtk=";
+
+    # packageJSON = ./package.json;
+    postBuild = ''
+      cp -r dist/ $out
+    '';
+  };
+
+  # copy the frontend static resources to final build directory
+  # Also modify tauri.conf.json so that it expects the resources at the new location
+  postPatch = ''
+    mkdir -p frontend
+    cp -r ${frontend}/* frontend
```

Comment:
in my opinion the `{}` make it easier read

Reply:
The `${var}` syntax is semantically different to `$var`

`${var}` means that Nix will substitute the variable during evaluation time and as you use `rec` here it will always be the attribute in this file.

`$var` is just the notation to let Bash substitute the variable, so its scope won't be the this nix file, but rather the (environment) variables set by stdenv.

The reason this is important is that `$var` will always have the actual final value of `var` while `${var}` (assuming rec is used) will only have the initial value.

Reply:
ah, that's new to me, thanks for clarifying